src/control/common/proto/mgmt/util.go

```diff
@@ -49,3 +56,20 @@ func (r *PoolSetPropReq) SetValueNumber(numVal uint64) {
 		Numval: numVal,
 	}
 }
+
+func (m *RanksReq) SetSystemRanks(sysRanks []system.Rank) error {
+	if m == nil {
+		return errors.New("nil request")
+	}
+	return convert.Types(sysRanks, &m.Ranks)
+}
+
+func (m *RanksReq) GetSystemRanks() []system.Rank {
+	if m != nil {
+		var sysRanks []system.Rank
+		if err := convert.Types(m.GetRanks(), &sysRanks); err == nil {
+			return sysRanks
+		}
+	}
+	return nil
+}
```

Comment:
Is it possible to avoid the duplication with SystemQueryReq by breaking some of this out into a helper method that could be shared? Something we could do with interfaces, maybe? I know it's not a huge amount of code but it's identical.

Reply:
"A little copying is better than a little dependency." ;) I think of these as the missing pieces to the generated code. I don't see a straightforward way to have the generator create these for us (maybe a custom plugin?), unfortunately.

Reply:
That's fair. It's not that big a deal.